apps/docs/components/Breadcrumbs.tsx

```diff
@@ -0,0 +1,159 @@
+'use client'
+
+import React, { Fragment } from 'react'
+import Link from 'next/link'
+import { usePathname } from 'next/navigation'
+import {
+  Breadcrumb_Shadcn_ as Breadcrumb,
+  BreadcrumbList_Shadcn_ as BreadcrumbList,
+  BreadcrumbItem_Shadcn_ as BreadcrumbItem,
+  BreadcrumbLink_Shadcn_ as BreadcrumbLink,
+  BreadcrumbSeparator_Shadcn_ as BreadcrumbSeparator,
+  BreadcrumbPage_Shadcn_ as BreadcrumbPage,
+  BreadcrumbEllipsis_Shadcn_ as BreadcrumbEllipsis,
+  Button,
+  cn,
+  Drawer,
+  DrawerClose,
+  DrawerContent,
+  DrawerFooter,
+  DrawerTrigger,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from 'ui'
+import { getMenuId } from '../app/guides/layout'
+import { useBreakpoint } from 'common'
+import * as NavItems from './Navigation/NavigationMenu/NavigationMenu.constants'
+
+const Breadcrumbs = ({ className }: { className?: string }) => {
+  const pathname = usePathname()
+  const menuId = getMenuId(pathname)
+  const menu = NavItems[menuId]
+  const breadcrumbs = findMenuItemByUrl(menu, pathname, [])
+  const [open, setOpen] = React.useState(false)
+  const isMobile = useBreakpoint('md')
+
+  const ITEMS_TO_DISPLAY = isMobile ? 4 : 3
+
+  if (!breadcrumbs?.length || breadcrumbs?.length === 1) return null
+
+  const appendedBreadcrumbs = breadcrumbs?.slice(-ITEMS_TO_DISPLAY + 1, isMobile ? -1 : undefined)
```

Comment:
@charislam @MildTomato I opted to omit the last breadcrumb on mobile and let intend that the title is the conclusion of the breadcrumb to save real estate and avoid too many multiple line breadcrumbs situations. Wdyt?

Desktop:
<img width="1340" alt="Screenshot 2024-07-12 at 19 19 08" src="https://github.com/user-attachments/assets/b24a8ae1-2564-4acb-b0aa-3f851c345fa1">

Mobile:
<img width="410" alt="Screenshot 2024-07-12 at 19 19 58" src="https://github.com/user-attachments/assets/0e22eb04-f4cf-4538-952e-078c621bbd33">

Reply:
Otherwise it would be like this.
I think we should optimise for mobile, this is too crowded.

<img width="413" alt="Screenshot 2024-07-12 at 19 22 34" src="https://github.com/user-attachments/assets/41435e7f-bf83-4cc5-9274-7b9e18d7937f">

Reply:
yea sure 👍